instrumentation/src/metrics_client.c

```diff
@@ -0,0 +1,108 @@
+#include "metrics_client.h"
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+static const int RECV_BUF_LEN = 1024;
+
+static char *const expected = "HTTP/1.1 200 OK";
+
+bool http_post(char *host, int port, char *method, char *path, char *postData);
+
+void send_otlp_metric() {
+    char *postData = "{\"resourceMetrics\":[{\"resource\":{},\"scopeMetrics\":[{\"scope\":{},\"metrics\":"
+                     "[{\"name\":\"splunk.linux-autoinstr.executions\",\"sum\":{\"dataPoints\":[{\"asInt\":\"1\"}],"
+                     "\"aggregationTemporality\":\"AGGREGATION_TEMPORALITY_DELTA\"}}]}]}]}";
+    char *host = "127.0.0.1";
+    int port = 4318;
+    char *method = "POST";
+    char *path = "/v1/metrics";
+    http_post(host, port, method, path, postData);
+}
+
+int make_socket(int timeout_seconds);
+
+bool connect_http(const char *host, int port, int socket_descriptor);
+
+bool post(int socket_descriptor, char *host, int port, char *method, char *path, char *postData);
+
+bool receive(int socket_descriptor);
+
+bool http_post(char *host, int port, char *method, char *path, char *postData) {
+    int socket_descriptor = make_socket(1);
+    if (socket_descriptor == -1) {
```

Comment:
The if statements on lines 38 and 45 use {} while the rest don't.  I'm not requesting a change, just wondering if there is a reason for it?